opentelemetry-sdk/src/metrics/aggregation.rs

```diff
@@ -0,0 +1,100 @@
+use std::fmt;
+
+use opentelemetry_api::metrics::{MetricsError, Result};
+
+/// The way recorded measurements are summarized.
+#[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub enum Aggregation {
+    /// An aggregation that drops all recorded data.
+    Drop,
+
+    /// An aggregation that uses the default instrument kind selection mapping to
+    /// select another aggregation.
+    ///
+    /// A metric reader can be configured to make an aggregation selection based on
+    /// instrument kind that differs from the default. This aggregation ensures the
+    /// default is used.
+    ///
+    /// See the [DefaultAggregationSelector] for information about the default
+    /// instrument kind selection mapping.
+    ///
+    /// [DefaultAggregationSelector]: crate::metrics::reader::DefaultAggregationSelector
+    Default,
+
+    /// An aggregation that summarizes a set of measurements as their arithmetic
+    /// sum.
+    Sum,
+
+    /// An aggregation that summarizes a set of measurements as the last one made.
+    LastValue,
+
+    /// An aggregation that summarizes a set of measurements as an histogram with
+    /// explicitly defined buckets.
+    ExplicitBucketHistogram {
+        /// The increasing bucket boundary values.
+        ///
+        /// Boundary values define bucket upper bounds. Buckets are exclusive of their
+        /// lower boundary and inclusive of their upper bound (except at positive
+        /// infinity). A measurement is defined to fall into the greatest-numbered
+        /// bucket with a boundary that is greater than or equal to the measurement. As
+        /// an example, boundaries defined as:
+        ///
+        /// vec![0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 1000.0];
+        ///
+        /// Will define these buckets:
+        ///
+        /// (-∞, 0], (0, 5.0], (5.0, 10.0], (10.0, 25.0], (25.0, 50.0], (50.0, 75.0],
+        /// (75.0, 100.0], (100.0, 250.0], (250.0, 500.0], (500.0, 1000.0], (1000.0, +∞)
+        boundaries: Vec<f64>,
+
+        /// Indicates whether to not record the min and max of the distribution.
+        ///
+        /// By default, these extrema are recorded.
+        ///
+        /// Recording these extrema for cumulative data is expected to have little
+        /// value, they will represent the entire life of the instrument instead of just
+        /// the current collection cycle. It is recommended to set this to true for that
+        /// type of data to avoid computing the low-value extrema.
+        no_min_max: bool,
```

Comment:
```suggestion
        record_min_max: bool,
```
To match spec wording:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation

Reply:
sure, will have to invert all the logic elsewhere too

Reply:
Ok done